### PR TITLE
Fix broken links before publish

### PR DESCRIPTION
**Pull request by ChrisBAshton**
Prevents publishing/scheduling of step by steps if they contain broken links.

Trello: https://trello.com/c/9hBzk8lJ/69-change-inset-text-to-tell-user-to-fix-broken-links

Screenshot:

Screen Shot 2019-09-20 at 13 46 05

### CODE REVIEW

File: spec/support/step_nav_steps.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 module StepNavSteps
+  include LinkChecker
```
[the remark this answers is not in the document]
Comment (ChrisBAshton) [reply spec/support/step_nav_steps.rb]: Interesting - if I leave it out I get errors:

```
  1) Managing step by step pages User publishes a page
     Failure/Error: stub_link_checker_report_success(step)
     
     NoMethodError:
       undefined method `stub_link_checker_report_success' for #<RSpec::ExampleGroups::ManagingStepByStepPages:0x000055ca82aca308>
       Did you mean?  stub_link_checker_api_check
     # ./spec/support/step_nav_steps.rb:95:in `given_there_is_a_step_by_step_page_with_a_link_report'
     # ./spec/features/managing_step_by_step_pages_spec.rb:52:in `block (2 levels) in <top (required)>'
```

[the remark this answers is not in the document]
Comment (brucebolt) [reply spec/support/step_nav_steps.rb]: Ah, of course.  `require` just loads the file, it doesn't actually do anything with the code contained within.

File: app/views/step_by_step_pages/show/_required_actions.html.erb [hunk 2]
```diff
@@ -15,7 +15,7 @@
         </li>
       <% end %>
 
-      <% if @step_by_step_page.steps.any? && !@step_by_step_page.links_checked_since_last_update? %>
+      <% if @step_by_step_page.steps.any? && !@step_by_step_page.links_checked_since_last_update? && !@step_by_step_page.broken_links_found? %>
```
[the remark this answers is not in the document]
Comment (ChrisBAshton) [reply app/views/step_by_step_pages/show/_required_actions.html.erb]: Silly question, but what sort of style should that be? Like this...?

```
<% if @step_by_step_page.steps.any? &&
      !@step_by_step_page.links_checked_since_last_update? &&
      !@step_by_step_page.broken_links_found? %>
  <li>
    <%= link_to "Check for broken links", '#steps', class: 'govuk-link govuk-link--no-visited-state' %>
  </li>
<% end %>
```

We don't have a linter for ERB files and I can't find any precedent for multiline conditionals in them. Having an internal debate as to whether the closing `%>` should be on its own line.

[the remark this answers is not in the document]
Comment (ChrisBAshton) [reply app/views/step_by_step_pages/show/_required_actions.html.erb]: Leaving as-is, as discussed at standup.

**Pull request merged**